src/firmware/app/primitives/move_manager.c

```diff
@@ -0,0 +1,224 @@
+    // onto its destination angle, so skip this if block
+    if (fabsf(pb->maj.disp) > APPROACH_LIMIT)
+    {
+        float wheel_axes[8];
+        build_wheel_axes(&wheel_axes, angle);
+        float theta_norm = atan2f(pb->pos[1], pb->pos[0]);
+        // use the pre-determined wheel axis
+        pb->rot.disp = min_angle_delta(wheel_axes[optimal_wheel_axes_index], theta_norm);
+    }
+}
+
+
+void plan_move_rotation(PhysBot* pb, float avel)
+{
+    float time_target = (pb->maj.time > TIME_HORIZON) ? pb->maj.time : TIME_HORIZON;
+    if (time_target > 0.5f)
+    {
+        time_target = 0.5f;
+    }
+    pb->rot.time  = time_target;
+    pb->rot.vel   = pb->rot.disp / pb->rot.time;
+    pb->rot.accel = (pb->rot.vel - avel) / TIME_HORIZON;
+    limit(&pb->rot.accel, MAX_T_A);
+}
+
+void app_move_manager_start(void* void_state_ptr, FirmwareWorld_t* world,
+                            MovePositionParams move_position_params, float final_angle)
+{
+    MoveManagerState_t* state = (MoveManagerState_t*)void_state_ptr;
+
+    // Convert into m/s and rad/s because physics is in m and s
+    state->destination[0] = move_position_params.destination.x_meters;
+    state->destination[1] = move_position_params.destination.y_meters;
+    state->destination[2] = final_angle;
+    state->end_speed      = move_position_params.final_speed_meters_per_second;
+    state->slow           = move_position_params.slow;
+
+    const FirmwareRobot_t* robot = app_firmware_world_getRobot(world);
+
+    float dx = state->destination[0] - app_firmware_robot_getPositionX(robot);
+    float dy = state->destination[1] - app_firmware_robot_getPositionY(robot);
+    // Add a small number to avoid division by zero
+    float total_disp    = sqrtf(dx * dx + dy * dy) + 1e-6f;
+    state->major_vec[0] = dx / total_disp;
+    state->major_vec[1] = dy / total_disp;
+    state->minor_vec[0] = state->major_vec[0];
+    state->minor_vec[1] = state->major_vec[1];
+    rotate(state->minor_vec, P_PI / 2);
+
+    // pick the wheel axis that will be used for faster movement
+    state->optimal_wheel_axes_index = choose_wheel_axis(
+        dx, dy, app_firmware_robot_getOrientation(robot), state->destination[2]);
+}
+
+void app_move_manager_end(void* void_state_ptr, FirmwareWorld_t* world) {}
+
+void app_move_manager_tick(void* void_state_ptr, FirmwareWorld_t* world)
+{
+    MoveManagerState_t* state = (MoveManagerState_t*)(void_state_ptr);
+    assert(!isnan(state->major_vec[0]));
+    assert(!isnan(state->major_vec[1]));
+    assert(!isnan(state->minor_vec[0]));
+    assert(!isnan(state->minor_vec[1]));
+    const FirmwareRobot_t* robot = app_firmware_world_getRobot(world);
+
+    PhysBot pb =
+        app_physbot_create(robot, state->destination, state->major_vec, state->minor_vec);
+
+    // choose a wheel axis to rotate onto
+    // TODO: try to make this less jittery
+    //    choose_rotation_destination(state->optimial_wheel_axes_index, &pb,
+    //    current_states.angle);
```

Comment:
Outstanding TODO comment.

Reply:
this was copied from the MovePrimitive, so fixing these things are out of scope